### ISSUE TITLE
VoiceOver Improvements

### ISSUE THREAD

**Issue by johanntan** [AudioBooth/AudioBooth#295]
There's a lot of images in the settings that VoiceOver reads through, I think this should be hidden from VoiceOver completely, it just adds more elements to swipe through when trying to get to a setting to adjust.
Also some of the settings, like let's say the naration speed default under player, should be better optomized to have that be an adjustable element to VO, right now a VoiceOver user will have to flick through the add button, then the actual speed, 1.0 for example, then the word ×. This should be condensed into simply something like "1.0×", with that being an adjustable element which you can then swipe up and down on with VO to set it.
The *Skip & Rewind* section also needs a lot of cleanup, there's a setting in there called *Rewind Range*, which I'm not sure if is adjustable or not, it doesn't read that way with VoiiceOver.

**Comment by aDENTinTIME** [comment]
A lot of these reported issues have been addressed in the recent beta. If you could take a look and let us know your thoughts that would be quite helpful.

**Comment by aDENTinTIME** [comment]
For my part, I found almost none of the menu images (icons) are read aloud now which is a massive improvement.

The narration speed settings are improved with the number being read out with the "times" symbol, and now you can swipe up and down to adjust it.

I found the skip & rewind section improved.

**Comment by aDENTinTIME** [comment]
I think under Settings > Home > Sections it would be cool if the sections were treated the same way as the sliders elsewhere as opposed to having the name and toggle separate; also for "Continue Listening" and "Continue Reading" maybe it would be good to combine "Always visible" like some of the other subtitles for switches.

**Comment by aDENTinTIME** [comment]
Under Settings > Storage the items could be grouped better for reading. Currently the total used is split so the number of GB is apart from the string "GB used".

**Comment by jeremygrenier** [comment]
I made several improvements on the last beta 1.11 (1783108261) if you want to give it a try.

**Comment by aDENTinTIME** [comment]
The “m” is reading out as “minutes” for me which is great!

**Comment by aDENTinTIME** [comment]
The narration speed controls seem to be just what was requested which is awesome.

**Comment by aDENTinTIME** [comment]
While poking around a book’s info page I noticed that metadata is read out smartly with info like “🗓️Published: 1995” being read all at once and skipping the icon. However, the read progress section’s time remaining and start date have the icon, delimiters, and info read independently. I imagine you can spot it easily and hopefully update it to read similarly to the metadata.

**Comment by johanntan** [comment]
HI,
So, I would like to propose adding of VoicieOver rotor actions to things in the app, books as an example.
This is a quick way for VO users to do something, for example by just swiping up and down on something, they could quickly select an ac tion, and execute it with a double-tap.
I would like these actions on books, because currently, trying to play from the continue listening section in the home tab still requires you to open the book details view then click continue listening.

* Play, if book is already playing then should of course change to pause
* Download/remove download
* an action to open the current book menu, just to get to it quicker from the book title.
Those are only my ideas for VO actions, if anyone else has any, they can probably comment it here as well.

Thanks

**Comment by aDENTinTIME** [comment]
It sounds like the actions you’re proposing overlap with the long-press action menu books have. I played around with VO for a minute and it seems if you select a book then tap twice, holding on the second, that will open the long-press menu with actions like play, download/remove and more. So I’m clear, is what you’re suggesting different? Are you suggesting another way to access these options _faster_, like tap on book then swipe down to the play option?

**Comment by johanntan** [comment]
> It sounds like the actions you’re proposing overlap with the long-press action menu books have.

Oh, yeah, I didn't even think of the long press menu, but yeah, for me it definitly is faster to just swipe down, double-tap play, then have the book playing.

**Comment by aDENTinTIME** [comment]
I gotcha, that’s a cool thought! I wonder what it’s like to implement.

**Comment by jonathans859** [comment]
Hi, another Voice Over user here. Adding Voice Over rotor actions shouldn't be to difficult to add. Glad that the settings issues have been addressed, I came here to report these too. Where could I find the beta to try it?

**Comment by jonathans859** [comment]
I found it. Under the skip and rewind settings, it would be cool if skip forward and back could also be adjustable sliders instead of a bunch of buttons you have to swipe through, would just be more intuetive and take less time.
Also, the buttons for the accent colour aren't labeled in the general section.
Voice Over actions for books, but also podcast episodes in the list would be amazing.
Also, on the podcast page for a specific episode, the button to download is labeled as arrow down circle for me, which should probably be called download. The play button is simply called x m left. For example 43m left button. It's not obvious you have to press it to play.
That's all for now. I'm looking forward to having these addressed, and thanks for making such a nice app.

**Comment by jeremygrenier** [comment]
I added the rotor actions and improvements mentioned above in the last beta if you want to give it a try.

**Comment by jonathans859** [comment]
Thanks much, it looks great. Nothing bad at all, but I find it quite interesting how you can change up the UI for Voice Over so significantly while it stays the same for sighted. I have a small vision rest and see that Voice Over just skips a lot of the clutter, while on screen these items are visible. Again, that's not a big problem in my opinion, as long as it does work. Just figured out I'd point it out since I noticed it.
